File: frontend/src/features/tax-profile/services/taxProfileService.ts
import api from "@/lib/api";

export async function getTaxProfile(clientId:number){
  const {data}=await api.get(`/tax-profiles/${clientId}`);
  return data;
}

export async function createTaxProfile(payload:any){
  const {data}=await api.post("/tax-profiles",payload);
  return data;
}

export async function updateTaxProfile(id:number,payload:any){
  const {data}=await api.put(`/tax-profiles/${id}`,payload);
  return data;
}

File: backend/app/services/tax_profile_service.py
from sqlalchemy.orm import Session

from app.models.tax_profile import TaxProfile
from app.schemas.tax_profile import TaxProfileCreate


def create_tax_profile(db: Session, data: TaxProfileCreate):
    profile = TaxProfile(**data.model_dump())
    db.add(profile)
    db.commit()
    db.refresh(profile)
    return profile


def get_tax_profile(db: Session, client_id: int):
    return (
        db.query(TaxProfile)
        .filter(TaxProfile.client_id == client_id)
        .first()
    )
from sqlalchemy.orm import Session
from app.models.tax_profile import TaxProfile

def update_tax_profile(db:Session,id:int,data:dict):

    profile=db.query(TaxProfile).filter(TaxProfile.id==id).first()

    for k,v in data.items():
        setattr(profile,k,v)

    db.commit()
    db.refresh(profile)

    return profile

def delete_tax_profile(db,id):

    profile=db.get(TaxProfile,id)

    db.delete(profile)

    db.commit()

    return {"message":"Tax Profile deleted"}

